### ISSUE TITLE
[java] new rule: CStyleArrayDeclaration

### ISSUE THREAD

**Issue by zbynek** [pmd/pmd#7026]
**Proposed Rule Name:** ArrayTypeStyle

**Proposed Category:** Code Style

**Description:**
Prefer Java style declaration of arrays as `int[] numbers` over C-style `int numbers[]`. The rule should apply both to fields and methods.

**Code Sample:** 

```java
class Numbers {
  double doubles[]; // bad
  int[] ints; // good
  double getDoubles()[] { // bad
    return new double[] { 1.0 };
  }

  int[] getInts() { // good
    return new int[] { 1 };
  }
}

```

**Possible Properties:**

None.

---

This can be already achieved using a trivial XPath rule `//VariableId/ArrayDimensions|//MethodDeclaration/ArrayDimensions`, but it would be nice to have it in PMD directly.

**Comment by pzygielo** [comment]
Would it also catch the case for methods?

```java
class P {
  double getDoubles()[] { // bad
    return new double[] { 1.0 };
  }

  int[] getInts() { // good
    return new int[] { 1 };
  }
}
```

**Comment by zbynek** [comment]
@pzygielo good point, description updated. 

Corresponding Checkstyle rule also covers both https://checkstyle.sourceforge.io/checks/misc/arraytypestyle.html
